Source/Rhetos.Deployment/ApplicationInitialization.cs

```diff
@@ -75,7 +76,14 @@ public static void ExecuteInitializer(IContainer container, Type initializerType
                         deployPackagesLogger.Trace("Initialization " + initializerType.Name + ".");
                         var initializers = initializerScope.Resolve<IPluginsContainer<IServerInitializer>>().GetPlugins();
                         IServerInitializer initializer = initializers.Single(i => i.GetType() == initializerType);
-                        initializer.Initialize();
+                        bool willExcute = WillExcuteInitilizerWithCondition(initializerType, initializerParams, out string message);

```

Comment:
Please use the `SkipRecompute` switch inside the `KeepSynchronizedRecomputeOnDeploy` class (see below the comment for DI), instead of skipping the whole initializer.

In KeepSynchronizedRecomputeOnDeploy, only one line of code should be skipped: `_genericRepositories.GetGenericRepository(keepSynchronized.Target).RecomputeFrom(keepSynchronized.Source);` and the warning should be reported for each skipped computation, so the developer can see which computations they need to execute manually. This solution also allows the `keepSyncRepos.Save(toInsert, toUpdate, toDelete);` line to be executed so that the server's metadata is up to date, otherwise the server would try to recompute the skipped data on next deployment it the switch is not used again.

Reply:
Resolved